### DISCUSSION TITLE
Can't change extension of downloaded file

### DISCUSSION THREAD

**Discussion by theromorph** [setvisible/DownZemAll#194] [Ideas]
Hello.

Sometimes I download some software from public catalog. One of today's link is https://plugring.farmanager.com/plugin.php?pid=974&l=en. I downloaded archives from 1 to 4.

1st issue: DZA does not intercept redirect or something and all bits and bytes go to "download.php" for every file, because of autorename I've got 4 "download.php" files like on the attached picture
![dza20201124-01](https://user-images.githubusercontent.com/6408649/100083279-45ec9900-2e5a-11eb-9d63-c81e41bc6a9c.jpg)

2nd issue: DZA does not properly rename files - it leaves file extension as is regardless my opinion. Because I know "real" file names of my "download*.php" files I tried to rename them, but result is somehow not the one I expected. The sequence I used:
select downloaded file in the list, press F2, paste new file name (or simply change the extension), press Enter. Result attached.
![dza20201124-02](https://user-images.githubusercontent.com/6408649/100084073-5c472480-2e5b-11eb-9298-e0e18d3fe05d.jpg)

p.s. I can't upgrade to 2.1.0 version, because downloaded installation file insists it has 2.0.8 version in it.

**Comment by setvisible** [comment]
Hi,
Thanks for reporting. Indeed I can reproduce the issue. 

At the moment, the app reads the URLs, but doesn't ask the server to know what it is (the actual resource, or a redirect).

I guess a good way to implement this feature could be that the app asks the server for each URL, the redirected resource details, in order to recover the real filename of the target resource. That is, DZA could display the final name "dirsync-x86.zip" instead of the wrong "download.php", and then solve the second problem (=renaming the extension), which wouldn't be useful to implement then :)

Be able to replace any URL ".../download.php?id=123" by the real resource name would be definitly a plus!

For the upgrade issue, I will open a separate ticket.

Regards
